### ISSUE TITLE
Pause and resume actions or not?

### ISSUE THREAD

**Issue by Pythocrates** [VDA5050/VDA5050#51]
## Context

The definition of the **startPause** action [6.8.1] states: "... Actions can continue...", whereas the definition of the **stopPause** action states: "... Movement and all other actions will be resumed (if any)..."
The "finished" column in the table in [6.8.2] suggests that actions should be paused (**startPause**: "... All actions will be paused..." and **stopPause**: "... All paused actions will be resumed...").

## Questions

1. Is it safe to assume that actions should be paused and that the first quote is a mistake?
1. If used in a preemptive fashion as described in [Issue #8](https://github.com/VDA5050/VDA5050/issues/8) (which makes perfect sense to me - if I want to pause something, I must preempt it), do we maybe need some additional flag indicating whether or not to resume any preempted actions?

**Comment by AntonDueck** [comment]
if you have atomic actions you may delay the action result until this action if finished, and the mark the pause action as finished. or you reject the pause action with an error.

so would prefer the first way.

**Comment by collani-bosch** [comment]
@AntonDueck : I agree; if an already running action cannot be paused, the state of 'startPause' is RUNNING, until the running action is finished. Rejecting 'startPause' would be unlucky, because it depends on timing whether  'startPause' is accepted or not.

**Comment by AntonDueck** [comment]
you have to inspect the action result (of all actions) after your instant action is finished to see what is been paused and which actions have finished.

**Comment by ConstantinUp** [comment]
In Version 3.0 we added the 'atomic' parameter to actions, determining, if pausing them is possible or not (#395). Adjusted the states in PR #450.
Please use PR #450 to comment on the concept, I added all of you as reviewers.
